packages/tests/server/react/useQuery.test.tsx

```diff
@@ -118,6 +118,25 @@ describe('useQuery()', () => {
     expectation.toMatchTypeOf<{ data: '__result' }>();
     expectation.not.toMatchTypeOf<{ data: undefined }>();
   });
+
+  test('data type with skipToken', () => {
+    const expectation = expectTypeOf(() =>
+      ctx.client.post.byId.useQuery(skipToken),
+    ).returns;
+
+    expectation.toMatchTypeOf<{ data: undefined }>();
+  });
+
+  test('data type with conditional skipToken', () => {
+    const expectation = expectTypeOf(() =>
+      ctx.client.post.byId.useQuery(
+        Math.random() > 0.5 ? skipToken : { id: '1' },
+      ),
+    ).returns;
+
+    expectation.toMatchTypeOf<{ data: '__result' | undefined }>();
+    expectation.not.toMatchTypeOf<{ data: undefined }>();
+  });
```

Comment:
I can also write exactly the same test for just useQuery(skipToken), it will have the exact same type as query can be populated imperatively. I think it's redundant, but let me know if you would like to have it